service/worker/scheduler/workflow.go

```diff
@@ -188,21 +188,34 @@ func (s *scheduler) run() error {
 			s.logger.Warn("Time went backwards", "from", t1, "to", t2)
 			t2 = t1
 		}
-		nextSleep := s.processTimeRange(
+		nextSleep, err := s.processTimeRange(
 			t1, t2,
 			// resolve this to the schedule's policy as late as possible
 			enumspb.SCHEDULE_OVERLAP_POLICY_UNSPECIFIED,
 			false,
 		)
+		if err != nil {
+			return err
+		}
 		s.State.LastProcessedTime = timestamp.TimePtr(t2)
 		// handle signals after processing time range that just elapsed
 		scheduleChanged := s.processSignals()
 		if scheduleChanged {
 			// need to calculate sleep again
-			nextSleep = s.processTimeRange(t2, t2, enumspb.SCHEDULE_OVERLAP_POLICY_UNSPECIFIED, false)
+			nextSleep, err = s.processTimeRange(t2, t2, enumspb.SCHEDULE_OVERLAP_POLICY_UNSPECIFIED, false)
+			if err != nil {
+				return err
+			}
 		}
 		// try starting workflows in the buffer
-		for s.processBuffer() {
+		for {
+			b, err := s.processBuffer()
+			if err != nil {
+				return err
```

Comment:
I appreciate the effort but this is wrong: returning an error here will end the workflow. Errors from terminate or cancel are supposed to be swallowed since they'll be retried at the next action attempt.

And most of the other changes in this file are just unnecessary clutter for type errors that can't ever happen.

Can we revert at least the changes to this file?